### DISCUSSION TITLE
VSG_ALPHA_TEST and vsgshadows example

### DISCUSSION THREAD

**Discussion by avilkinnick** [vsg-dev/VulkanSceneGraph#1308] [Q&A] [answered]
Hi, I copied VSG example "vsgshadows" and trying to load my gltf model through this, it is loading without extra arguments, but when I add argument "--at" to add alpha channel to textures, such error is shown:
```
Warning: GLSL source failed to parse.
Warning: glslang info log:
ERROR: 0:1010: 'material' : undeclared identifier 
ERROR: 0:1010: 'alphaMask' : vector swizzle too long 
ERROR: 0:1010: 'alphaMask' : unknown swizzle selection 
ERROR: 0:1010: 'alphaMask' : unknown swizzle selection 
ERROR: 0:1010: 'alphaMask' : vector swizzle selection out of range 
ERROR: 0:1010: '' : missing #endif 
ERROR: 0:1010: '' : compilation terminated 
ERROR: 7 compilation errors.  No code generated.
```

**Comment by robertosfield** [comment, marked as answer]
I have just checked vsgshadow --at with the default model and gltf models and it's running cleanly for me.  I'm using VSG, vsgXchange and vsgExample master on Kubuntu 24.04.

What version of VSG, vsgXchange and vsgExamples are you using?  Is the VSG built with glslang support?

**Reply by avilkinnick** [reply]
Thanks, I just needed new shaders from vsgExamples
